www/js/diary/detail.js

```diff
@@ -0,0 +1,136 @@
+      tileLayerOptions: {
+          opacity: 0.9,
+          detectRetina: true,
+          reuseTiles: true,
+      }
+    } 
+  });
+  $scope.getIcon = function(section) {
+    var icons = {"BICYCLING":"ion-android-bicycle",
+                  "WALKING":" ion-android-walk",
+                  "RUNNING":" ion-android-walk",
+                  "IN_VEHICLE":"ion-disc",}
+    return icons[$scope.getHumanReadable(section.properties.sensed_mode)]; 
+  }
+ $scope.getHumanReadable = function(sensed_mode) {
+      ret_string = sensed_mode.split('.')[1];
+      if(ret_string == 'ON_FOOT') {
+          return 'WALKING';
+      } else {
+          return ret_string;
+      }
+    };
+    $scope.getPercentages = function(trip) {
+      var rtn0 = []; // icons
+      var rtn1 = []; //percentages
+      var icons = {"BICYCLING":"ion-android-bicycle",
+                    "WALKING":" ion-android-walk",
+                    "RUNNING":" ion-android-walk",
+                    "IN_VEHICLE":"ion-disc",}
+      var total = 0;
+      for (var i=0; i<trip.sections.length; i++) {
+        if (rtn0.indexOf($scope.getHumanReadable(trip.sections[i].properties.sensed_mode)) == -1) {
+          rtn0.push($scope.getHumanReadable(trip.sections[i].properties.sensed_mode));
+          rtn1.push(trip.sections[i].properties.distance);
+          total += trip.sections[i].properties.distance;
+        } else {
+          rtn1[rtn0.indexOf($scope.getHumanReadable(trip.sections[i].properties.sensed_mode))] += trip.sections[i].properties.distance;
+          total += trip.sections[i].properties.distance;
+        }
+      }
+      for (var i=0; i<rtn0.length; i++) {
+        rtn0[i] = "icon " + icons[rtn0[i]];
+        rtn1[i] = Math.floor((rtn1[i] / total) * 100);
+      }
+      return [rtn0, rtn1];
+    }
+
+    $scope.starColor = function(trip) {
+      if (trip.common_count >= 3) {
+        return 'yellow';
+      } else {
+        return 'transparent';
+      }
+    }
+    $scope.allModes = function(trip) {
+      var rtn = [];
+      var icons = {"BICYCLING":"ion-android-bicycle",
+                    "WALKING":" ion-android-walk",
+                    "RUNNING":" ion-android-walk",
+                    "IN_VEHICLE":"ion-disc",}
+      for (var i=0; i<trip.sections.length; i++) {
+        if (rtn.indexOf($scope.getHumanReadable(trip.sections[i].properties.sensed_mode)) == -1) {
+          rtn.push($scope.getHumanReadable(trip.sections[i].properties.sensed_mode));
+        }
+      }
+      for (var i=0; i<rtn.length; i++) {
+        rtn[i] = "icon " + icons[rtn[i]];
+      }
+
+      return rtn;
+    }
+
+  $scope.trip = Timeline.getTrip($stateParams.tripId);
+        $scope.getKmph = function(section) {
+            metersPerSec = section.properties.distance / section.properties.duration;
+            return (metersPerSec * 3.6).toFixed(2);
+        };
+
+        $scope.getFormattedDistance = function(dist_in_meters) {
```

Comment:
As I said in email, please pull these into a common.js that is used by both list and detail views. Duplicated code is both buggy and ugly.